cmd/chart-repo/utils.go

```diff
@@ -98,6 +98,10 @@ func syncRepo(dbSession datastore.Session, repoName, repoURL string, authorizati
 	}
 
 	charts := chartsFromIndex(index, r)
+	if len(charts) == 0 {
+		log.Fatal("No charts are found")
```

Comment:
can I suggest simply returning an `errors.New("no charts in repository index")`, as the caller of this function will log the error with Fatal: https://github.com/helm/monocular/blob/e19c1b1818e0e611257db71247f7d08705ac42f2/cmd/chart-repo/sync.go#L64.

The issue with using Fatal here is that it will exit the program here, and that should really be the job of the caller of this function. This function should just return a response and not control the flow of the program.

Reply:
@prydonius Thanks for explaining. 

Do you mean to add a test something like this: 
``` 
type chartHTTPClient struct{}

func (h *chartHTTPClient) Do(req *http.Request) (*http.Response, error){
	w := httptest.NewRecorder()
	// Don't accept trailing slashes
	if strings.HasPrefix(req.URL.Path, "//") {
		w.WriteHeader(500)
	}
	// If subpath repo URL test, check that index.yaml is correctly added to the
	// subpath
	if req.URL.Host == "subpath.test" && req.URL.Path != "/subpath/index.yaml" {
		w.WriteHeader(500)
	}
	w.Write(nil)
	return w.Result(), nil
}

Reply:
@hpandeycodeit I think we can remove the trailing slashes and subpath parts in that, and just return an empty index. We should also call this something like `emptyChartRepoHTTPClient` to make the intent more clear.

I'm also not sure if `w.Write(nil)` will be sufficient though, we might need to create a test index file (like https://github.com/helm/monocular/blob/e19c1b1818e0e611257db71247f7d08705ac42f2/cmd/chart-repo/testdata/valid-index.yaml#L1) that has an `entries` key that is empty.

e.g.:

```
type emptyChartRepoHTTPClient struct{}

func (h *emptyChartRepoHTTPClient) Do(req *http.Request) (*http.Response, error){
	w := httptest.NewRecorder()

        w.Write([]byte(emptyRepoIndexYAML))
	return w.Result(), nil
}
```

We would then need a test that is similar to https://github.com/helm/monocular/blob/e19c1b1818e0e611257db71247f7d08705ac42f2/cmd/chart-repo/utils_test.go#L157, but also sets the `netClient` like https://github.com/helm/monocular/blob/e19c1b1818e0e611257db71247f7d08705ac42f2/cmd/chart-repo/utils_test.go#L202, and asserts than an error is returned.

Does that make sense?

Reply:
@prydonius Yes, it does. I have added the test case. Let me know if it looks okay to you. 

:)